src/distributed_planner/plan_annotator.rs

```diff
@@ -300,3 +299,252 @@ fn required_network_boundary_below(parent: &dyn ExecutionPlan) -> Option<Require
+     Temp9am [type=DOUBLE] [repetitiontype=OPTIONAL]
+     Temp3pm [type=DOUBLE] [repetitiontype=OPTIONAL]
+     RainToday [type=BYTE_ARRAY] [convertedtype=UTF8] [repetitiontype=OPTIONAL]
+     RISK_MM [type=DOUBLE] [repetitiontype=OPTIONAL]
+     RainTomorrow [type=BYTE_ARRAY] [convertedtype=UTF8] [repetitiontype=OPTIONAL]
+    */
+
+    #[tokio::test]
+    async fn test_select_all() {
+        let query = r#"
+        SELECT * FROM weather
+        "#;
+        let annotated = sql_to_annotated(query).await;
+        assert_snapshot!(annotated, @"DataSourceExec: task_count=Desired(3)")
+    }
+
+    #[tokio::test]
+    async fn test_aggregation() {
+        let query = r#"
+        SELECT count(*), "RainToday" FROM weather GROUP BY "RainToday" ORDER BY count(*)
+        "#;
+        let annotated = sql_to_annotated(query).await;
+        assert_snapshot!(annotated, @r"
+        ProjectionExec: task_count=Maximum(1)
+          SortPreservingMergeExec: task_count=Maximum(1), required_network_boundary=Coalesce
+            SortExec: task_count=Desired(2)
+              ProjectionExec: task_count=Desired(2)
+                AggregateExec: task_count=Desired(2)
+                  CoalesceBatchesExec: task_count=Desired(2), required_network_boundary=Shuffle
+                    RepartitionExec: task_count=Desired(3)
+                      RepartitionExec: task_count=Desired(3)
+                        AggregateExec: task_count=Desired(3)
+                          DataSourceExec: task_count=Desired(3)
+        ")
+    }
+
+    #[tokio::test]
+    async fn test_left_join() {
+        let query = r#"
+        SELECT a."MinTemp", b."MaxTemp" FROM weather a LEFT JOIN weather b ON a."RainToday" = b."RainToday"
+        "#;
+        let annotated = sql_to_annotated(query).await;
+        assert_snapshot!(annotated, @r"
+        CoalesceBatchesExec: task_count=Maximum(1)
+          HashJoinExec: task_count=Maximum(1)
+            CoalescePartitionsExec: task_count=Maximum(1)
+              DataSourceExec: task_count=Maximum(1)
+            DataSourceExec: task_count=Maximum(1)
+        ")
+    }
+
+    #[tokio::test]
+    async fn test_left_join_distributed() {
+        let query = r#"
+        WITH a AS (
+            SELECT
+                AVG("MinTemp") as "MinTemp",
+                "RainTomorrow"
+            FROM weather
+            WHERE "RainToday" = 'yes'
+            GROUP BY "RainTomorrow"
+        ), b AS (
+            SELECT
+                AVG("MaxTemp") as "MaxTemp",
+                "RainTomorrow"
+            FROM weather
+            WHERE "RainToday" = 'no'
+            GROUP BY "RainTomorrow"
+        )
+        SELECT
+            a."MinTemp",
+            b."MaxTemp"
+        FROM a
+        LEFT JOIN b
+        ON a."RainTomorrow" = b."RainTomorrow"
+        "#;
+        let annotated = sql_to_annotated(query).await;
+        assert_snapshot!(annotated, @r"
+        CoalesceBatchesExec: task_count=Maximum(1)
```

Comment:
This might be a good example for a doc comment somewhere. It shows the scale factor working, Maximum(1) for hash join, and `propagate_task_count` working together.